parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -357,43 +373,31 @@ retrycmd_curl_file() {
         fi
     done
 }
-systemctl_restart() {
-    retries=$1; wait_sleep=$2; timeout=$3 svcname=$4
+# base systemctl retry command, should not be called directly - use systemctl_restart, systemctl_stop, systemctl_disable
+_systemctl_retry_svc_operation() {
+    retries=$1; wait_sleep=$2; timeout=$3 operation=$4 svcname=$5 shouldLogRetryInfo=${6:-false}
     for i in $(seq 1 $retries); do
         timeout $timeout systemctl daemon-reload
-        timeout $timeout systemctl restart $svcname && break || \
+        timeout $timeout systemctl $operation $svcname && break || \
         if [ $i -eq $retries ]; then
             return 1
         else
-            systemctl status $svcname --no-pager -l
-            journalctl -u $svcname
+          if [[ "$shouldLogRetryInfo" == "true"  ]]; then
```

Comment:
we're gonna be enabling SC rules to catch thus, but for now:

```suggestion
          if [ "$shouldLogRetryInfo" = "true"  ]; then
```

Reply:
cc @Devinwong